src/app/tracing/graph/cy-graph/cy-listeners.ts

```diff
@@ -82,6 +93,36 @@ export function addCyZoomListener(cy: Cy, onZoom: () => void): void {
     cy.on(CY_EVENT_ZOOM, onZoom);
 }
 
+export function addCyBoxZoomListerner(
+    cy: Cy,
+    onBoxZoom: (boxStartPosition: Position, boxEndPosition: Position) => void,
+): void {
+    let lastBoxStart: BoxStartProps | null = null;
+
+    cy.on(CY_EVENT_BOX_START, (event: CyMouseEvent) => {
+        // Checking both ctrl and meta key because Windows & Linux generally use ctrl, while Mac uses the Cmd key.
+        const isBoxZoom =
+            event.originalEvent.ctrlKey || event.originalEvent.metaKey;
+        lastBoxStart = {
+            isBoxZoom: isBoxZoom,
+            position: event.position,
+        };
+        if (isBoxZoom) {
+            cy.elements().unselectify();
+        }
+    });
+
+    cy.on(CY_EVENT_BOX_END, (event: CyMouseEvent) => {
+        if (lastBoxStart?.isBoxZoom && lastBoxStart?.position) {
+            onBoxZoom(lastBoxStart.position, event.position);
+        }
+        lastBoxStart = null;
+        window.setTimeout(() => {
```

Comment:
just out of interest: why do we need a timeout here?

Reply:
If we simply selectify() the graph elements again inside this function, then the box select will still select elements in addition too zooming in. 
Calling selectify() in a setTimeout causes it to happen after this event has bubbled up the chain and the boxSelect and boxEnd events have been handled by cy.

I would prefer a more semantic solution, but I'm not sure how else to cause the effect we are looking for here.